### ISSUE TITLE
Limiting RHL (Winter is coming)

### ISSUE THREAD

**Issue by theKashey** [gaearon/react-hot-loader#991]
User request:
```js
import { nothot } from "react-hot-loader";
import GraphiQL as GQL from "graphiql";

const GraphiQL = nothot(GQL);  // RHL will not apply proxies to this function
...
```

Doable, but we could improve it - for example not proxy any components from node_modules, as long they should be considered static. (see https://github.com/gaearon/react-hot-loader/issues/304#issuecomment-369495574)

Way to reproduce:
1. Extend babel plugin to node_modules.
2. Now RHL knew components origin
3. `hothow` to all components with origin in node_modules

**Comment by theKashey** [comment]
- A new ❄️ API - just `cold`, in the opposite of `hot`.
- hook inside reactHotLoader.register(babelPlugin) to cool components by file patterns.

**Comment by ljharb** [comment]
It wouldn’t be reasonable to ask components that need to not be hot, to add a dep on react-got-loader.

Any mechanism for this should be something that doesn’t require adding this dep.

**Comment by neoziro** [comment]
I love the idea!

**Comment by theKashey** [comment]
@ljharb - by default all _your_ components _could_ be changed and hot-reloaded, and all _thirdparty_ could not be changed, and thus we probably shall disable RHL for them, to lower side effects (type comparison, for example).
As result cold API gives your ability to "cold" components you "know" and get reference, and "cold" anything else by file mask.

The only problem - it will not work out of the box, cos 99% babel setups excludes node_modules.

**Comment by ljharb** [comment]
@thekashey yes, that’s true. But if hot loader won’t choose to be safe out of the box, then it seems right to offer an output that minimally inconveniences third parties.

**Comment by theKashey** [comment]
I am just not activating this untested feature out of the box, _yet_. And that is ok, as long it does not change the default RHL behavior.

**Comment by theKashey** [comment]
fixed in v4.3.0
